### PR TITLE
ETL-3519: Add mysql utils to gobblin-utility

### PR DESCRIPTION
**Pull request by ydai1124**
Added jdbc module to gobblin-util.
Deprecated DataSourceProvider in gobblin-metastore.
@liyinan926 @chavdar can you review?

### CODE REVIEW

File: gobblin-metastore/src/main/java/gobblin/metastore/DataSourceProvider.java [hunk 1]
```diff
@@ -12,42 +12,21 @@
...
- *
- * @author ynli
+ * {@inheritDoc}
+ * @deprecated This class is replaced by {@link JobHistoryDataSourceProvider}
  */
-public class DataSourceProvider implements Provider<DataSource> {
-
-  private final BasicDataSource basicDataSource;
+@Deprecated
+public class DataSourceProvider extends JobHistoryDataSourceProvider {
```
Comment (liyinan926) [inline gobblin-metastore/src/main/java/gobblin/metastore/DataSourceProvider.java]: Actually you don't need to keep this and deprecate as this class is not part of the public API.

File: gobblin-metastore/src/main/java/gobblin/metastore/JobHistoryDataSourceProvider.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+
+/**
+ * This class extends {@link gobblin.util.jdbc.DataSourceProvider} with its own property keys.
+ */
+public class JobHistoryDataSourceProvider extends gobblin.util.jdbc.DataSourceProvider {
+
+  @Inject
+  public JobHistoryDataSourceProvider(@Named("dataSourceProperties") Properties properties) {
+    super();
```
Comment (liyinan926) [inline gobblin-metastore/src/main/java/gobblin/metastore/JobHistoryDataSourceProvider.java]: This is not necessary as it's called implicitly.

File: gobblin-utility/src/main/java/gobblin/util/jdbc/DataSourceModule.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * A Guice module defining the dependencies used by the jdbc data source.
+ */
+public class DataSourceModule extends AbstractModule {
+
+  private final Properties properties;
+
+  public DataSourceModule(Properties properties) {
+    this.properties = properties;
```
Comment (liyinan926) [inline gobblin-utility/src/main/java/gobblin/util/jdbc/DataSourceModule.java]: I suggest making a copy here so any potential updates to the `properties` won't affect the behavior of this module.

**Comment by ydai1124** [comment]
@liyinan926 Updated.

**Comment by chavdar** [comment]
LGTM

**Pull request merged**